cvat/apps/iam/views.py

```diff
@@ -98,14 +100,40 @@ def post(self, request, *args, **kwargs):
 class RegisterViewEx(RegisterView):
     def get_response_data(self, user):
         data = self.get_serializer(user).data
-        data['email_verification_required'] = True
-        data['key'] = None
         if allauth_settings.EMAIL_VERIFICATION != \
             allauth_settings.EmailVerificationMethod.MANDATORY:
             data['email_verification_required'] = False
             data['key'] = user.auth_token.key
         return data
 
+    # NOTE: we should reimplement this method to fix the following issue:
+    # In the previous used version of dj-rest-auth 2.2.7, if the REST_SESSION_LOGIN setting was not defined in the settings file,
+    # the default value specified in the documentation (https://dj-rest-auth.readthedocs.io/en/2.2.7/configuration.html)
+    # was not applied for some unknown reason, and an authentication token was added to a user.
+    # With the dj-rest-auth version 5.0.2, there have been changes to how settings are handled,
+    # and now the default value is properly taken into account.
+    # However, even with the updated code, it still does not handle the scenario
+    # of handling two authentication flows simultaneously during registration process.
+    # Since there is no mention in the dj-rest-auth documentation that session authentication
+    # cannot be used alongside token authentication (https://dj-rest-auth.readthedocs.io/en/latest/configuration.html),
+    # and given the login implementation (https://github.com/iMerica/dj-rest-auth/blob/c6b6530eb0bfa5b10fd7b9e955a39301156e49d2/dj_rest_auth/views.py#L69-L75),
+    # this situation appears to be a bug.
```

Comment:
Can you submit an issue/PR to the upstream project?

Reply:
Your PR seems to be failing the tests. Can you fix it?

Reply:
Yes, I'll fix them, but honestly, tests failing is not related to this fix

Reply:
Why are they failing, then?

Reply:
Because after [overriding](https://github.com/iMerica/dj-rest-auth/blob/6807ea06a9ce0f4f11d607d0005f5d99aabe5f99/dj_rest_auth/tests/test_api.py#L532) dj rest auth internal settings, the class attribute [token_model](https://github.com/iMerica/dj-rest-auth/blob/6807ea06a9ce0f4f11d607d0005f5d99aabe5f99/dj_rest_auth/registration/views.py#L37) is not overridden and remains the [default value](https://github.com/iMerica/dj-rest-auth/blob/6807ea06a9ce0f4f11d607d0005f5d99aabe5f99/dj_rest_auth/app_settings.py#L23)

Reply:
If you change the default value of `TOKEN_MODEL`, the test will pass

Reply:
@SpecLad, now the tests are passed.